Feature/radar chart vertice label

Koichi5 opened the pull request:
## Summary
Added a new feature to display labels at each vertex of the RadarChart. 
Fixes this issue: https://github.com/imaNNeo/fl_chart/issues/1886

## What's included
### New Classes and Methods
- Added `RadarChartVerticeLabel` class to handle vertex label configuration
  - `text`: Displays the label text
  - `positionPercentageOffset`: Controls label position relative to vertex

### Tests
Added comprehensive test cases for `drawVerticeLabels`:
1. Test case for when no vertex labels are configured
2. Test case for basic vertex label display
3. Test case for custom position percentage offset

## Screenshot
<img width="372" alt="スクリーンショット 2025-05-01 23 47 26" src="https://github.com/user-attachments/assets/88a00c14-d83d-41ad-b030-f6e55b8eb102" />

Koichi5 committed:
feature: Add vertice label feature

Koichi5 committed:
feature: Add vertice label tests

imaNNeo commented:
I didn't get what the difference between `positionPercentageOffset` and `verticeLabelPositionPercentageOffset` is?

imaNNeo commented:
And your sample looks beautiful. Can you please add it in our samples list?

Koichi5 commented:
Thank you for your feedback!

After reconsidering, I realized that `positionPercentageOffset` and `verticeLabelPositionPercentageOffset` essentially serve the same purpose. Therefore, I would like to remove `verticeLabelPositionPercentageOffset` for clarity and simplicity.

The role of `positionPercentageOffset` in `RadarChartVerticeLabel` is the same as in the `RadarChartTitle` method: it controls the distance between each vertex and its corresponding label.

| `positionPercentageOffset` in `RadarChartVerticeLabel` = 0 |`positionPercentageOffset` in `RadarChartVerticeLabel` = 1 |
|--------|--------|
| <img src="https://github.com/user-attachments/assets/39c2f1a1-7df6-49a8-bbe4-b144ed868299" width=300px> | <img src="https://github.com/user-attachments/assets/800a2f7f-2ae7-48ce-a815-214620c4d39d" width=300px> | 



I will also add the sample to the samples list as you suggested!

Koichi5 committed:
delete: Delete verticeLabelPositionPercentageOffset in RadarChartData

Koichi5 committed:
feature: Add RadarChart sample2 in samples list

Koichi5 commented:
@imaNNeo 
Hi! I’ve added the Radar Chart sample and removed the verticalLabelPositionPercentageOffset as discussed.  
Whenever you get a chance, I’d appreciate a quick review. Thanks!

imaNNeo commented on lib/src/chart/radar_chart/radar_chart_data.dart:
```diff
@@ -130,9 +152,15 @@ class RadarChartData extends BaseChartData with EquatableMixin {
   final TextStyle? titleTextStyle;
 
+  /// Defines style of showing [RadarChart] vertice labels.
+  final TextStyle? verticeLabelTextStyle;
```
It seems you forgot to add this `verticeLabelTextStyle` property in the copyWith and lerp functions

imaNNeo commented on lib/src/chart/radar_chart/radar_chart_data.dart:
```diff
@@ -41,6 +45,22 @@ class RadarChartTitle {
 }
 
+/// Defines a label for [RadarChart] vertices
+class RadarChartVerticeLabel {
```
There's a typo here, and this is what I've got from the Chatgpt:
<img width="1029" height="685" alt="image" src="https://github.com/user-attachments/assets/a1f91c5b-17f2-46d6-9c31-551b61db67bc" />

Koichi5 committed:
fix: Fix typo and change property name

imaNNeo commented:
Sorry, I still see some typos:
<img width="300" alt="image" src="https://github.com/user-attachments/assets/2e66ba41-c86b-456e-809b-6a77151823d0" />

<img width="300" alt="image" src="https://github.com/user-attachments/assets/cf3b31be-101d-4377-ba65-33a0b9dad1f5" />

imaNNeo commented:
Closed due to inactivity and a lot of conflicts
Feel free to create a new one (or re-open this one with the needed changes)
